ingestion/src/metadata/profiler/profiler/datalake_sampler.py

```diff
@@ -50,53 +47,28 @@ def __init__(
     def _fetch_rows(self, data_frame):
         from pandas import notnull  # pylint: disable=import-outside-toplevel
 
-        sampled_data_frame = data_frame.sample(
-            n=(int(self.profile_sample) or 100)
-            if self.profile_sample_type == ProfileSampleType.ROWS
-            else None,
-            frac=self.profile_sample
-            if self.profile_sample_type == ProfileSampleType.PERCENTAGE
-            else None,
-            random_state=random.randint(0, 100),
-            replace=True,
-        )
         return (
-            sampled_data_frame.astype(object)
+            data_frame.astype(object)
             .where(
-                notnull(sampled_data_frame),
+                notnull(data_frame),
                 None,
             )
-            .values.tolist()
+            .values.tolist()[:100]
         )
 
     def get_col_row(self, data_frame):
         """
         Fetches columns and rows from the data_frame
         """
-        from pandas import DataFrame  # pylint: disable=import-outside-toplevel
-
         cols = []
-        chunk = None
-        if isinstance(data_frame, DataFrame):
-            return (
-                data_frame.columns.tolist(),
-                self._fetch_rows(data_frame),
-            )
-        chunk_limit = math.ceil(self.profile_sample / CHUNKSIZE)
-        cols = data_frame[0].columns.tolist()
         rows = []
-        for index, chunk in enumerate(data_frame):
-            if index >= chunk_limit:
+        cols = data_frame[0].columns.tolist()
+        for chunk in data_frame:
+            rows.extend(self._fetch_rows(chunk)[: 100 - len(rows)])
```

Comment:
same here, shouldn't it be `self.sample_limit` vs 100

Reply:
same, I have modified the logic, and made it more readable, added a comment as well